fv3core/stencils/fv_dynamics.py

```diff
@@ -253,65 +182,254 @@ def wrapup(state, comm: CubedSphereCommunicator):
+        ArgSpec("uc", "x_wind_on_c_grid", "m/s", intent="inout"),
+        ArgSpec("vc", "y_wind_on_c_grid", "m/s", intent="inout"),
+        ArgSpec("q_con", "total_condensate_mixing_ratio", "kg/kg", intent="inout"),
+        ArgSpec("pe", "interface_pressure", "Pa", intent="inout"),
+        ArgSpec("phis", "surface_geopotential", "m^2 s^-2", intent="in"),
+        ArgSpec(
+            "pk",
+            "interface_pressure_raised_to_power_of_kappa",
+            "unknown",
+            intent="inout",
+        ),
+        ArgSpec(
+            "pkz",
+            "layer_mean_pressure_raised_to_power_of_kappa",
+            "unknown",
+            intent="inout",
+        ),
+        ArgSpec("ps", "surface_pressure", "Pa", intent="inout"),
+        ArgSpec("omga", "vertical_pressure_velocity", "Pa/s", intent="inout"),
+        ArgSpec("ak", "atmosphere_hybrid_a_coordinate", "Pa", intent="in"),
+        ArgSpec("bk", "atmosphere_hybrid_b_coordinate", "", intent="in"),
+        ArgSpec("mfxd", "accumulated_x_mass_flux", "unknown", intent="inout"),
+        ArgSpec("mfyd", "accumulated_y_mass_flux", "unknown", intent="inout"),
+        ArgSpec("cxd", "accumulated_x_courant_number", "", intent="inout"),
+        ArgSpec("cyd", "accumulated_y_courant_number", "", intent="inout"),
+        ArgSpec(
+            "diss_estd",
+            "dissipation_estimate_from_heat_source",
+            "unknown",
+            intent="inout",
+        ),
+    )
+
+    def __init__(self, comm: fv3gfs.util.CubedSphereCommunicator, namelist):
+        self.comm = comm
+        self.grid = spec.grid
+        self.namelist = namelist
+        self.do_halo_exchange = global_config.get_do_halo_exchange()
+        self.tracer_advection = fv3core.stencils.tracer_2d_1l.Tracer2D1L(comm, namelist)
+        # npx and npy are number of interfaces, npz is number of centers
+        # and shapes should be the full data shape
+        self._temporaries = fvdyn_temporaries(
+            self.grid.domain_shape_full(add=(1, 1, 1))
+        )
+        if not (not self.namelist.inline_q and DynamicalCore.NQ != 0):
+            raise NotImplementedError("tracer_2d not implemented, turn on z_tracer")
+
+    def step_dynamics(
+        self,
+        state: Mapping[str, fv3gfs.util.Quantity],
+        consv_te,
+        do_adiabatic_init: bool,
+        timestep: float,
+        ptop,
+        n_split: int,
+        ks,
+        timer: fv3gfs.util.Timer = fv3gfs.util.NullTimer(),
+    ):
+        """
+        Step the model state forward by one timestep.
+
+        Args:
+            state: model prognostic state and inputs
+            consv_te: ???
+            do_adiabatic_init: if True, do adiabatic dynamics. Used
+                for model initialization.
+            timestep: time to progress forward in seconds
+            ptop: pressure at top of atmosphere
+            n_split: number of acoustic timesteps per remapping timestep
+            ks: ???
```

Comment:
this is to set the bottom layer for which rayleigh friction and other rayliegh computations are done